controller_interface/include/semantic_components/gps_sensor.hpp

```diff
@@ -0,0 +1,130 @@
+#define SEMANTIC_COMPONENTS__GPS_SENSOR_HPP_
+
+#include <array>
+#include <string>
+#include <vector>
+
+#include "semantic_components/semantic_component_interface.hpp"
+#include "sensor_msgs/msg/nav_sat_fix.hpp"
+
+namespace semantic_components
+{
+
+template <bool use_covariance>
+class GPSSensor : public SemanticComponentInterface<sensor_msgs::msg::NavSatFix>
+{
+public:
+  explicit GPSSensor(const std::string & name)
+  : SemanticComponentInterface(
+      name, {{name + "/" + "status"},
+             {name + "/" + "service"},
+             {name + "/" + "latitude"},
+             {name + "/" + "longitude"},
+             {name + "/" + "altitude"}})
+  {
+    if constexpr (use_covariance)
+    {
+      interface_names_.emplace_back(name + "/" + "latitude_covariance");
+      interface_names_.emplace_back(name + "/" + "longitude_covariance");
+      interface_names_.emplace_back(name + "/" + "altitude_covariance");
+      covariance_.fill(0.0);
+    }
+  }
+
+  /**
+   * Return GPS's status e.g. fix/no_fix
+   *
+   * \return Status
+   */
+  int8_t get_status() const { return static_cast<int8_t>(state_interfaces_[0].get().get_value()); }
+
+  /**
+   * Return service used by GPS e.g. fix/no_fix
+   *
+   * \return Service
+   */
+  uint16_t get_service() const
+  {
+    return static_cast<uint16_t>(state_interfaces_[1].get().get_value());
+  }
+
+  /**
+   * Return latitude reported by a GPS
+   *
+   * \return Latitude.
+   */
+  double get_latitude() const { return state_interfaces_[2].get().get_value(); }
+
+  /**
+   * Return longitude reported by a GPS
+   *
+   * \return Longitude.
+   */
+  double get_longitude() const { return state_interfaces_[3].get().get_value(); }
+
+  /**
+   * Return altitude reported by a GPS
+   *
+   * \return Altitude.
+   */
+  double get_altitude() const { return state_interfaces_[4].get().get_value(); }
+
+  /**
+   * Return covariance reported by a GPS.
+   *
+   * \return Covariance array.
+   */
+  template <typename U = void, typename = std::enable_if_t<use_covariance, U>>
+  const std::array<double, 9> & get_covariance()
+  {
+    covariance_[0] = state_interfaces_[5].get().get_value();
+    covariance_[4] = state_interfaces_[6].get().get_value();
+    covariance_[8] = state_interfaces_[7].get().get_value();
+    return covariance_;
+  }
+
+  /**
+   * Fills a NavSatFix message from the current values.
+   */
+  bool get_values_as_message(sensor_msgs::msg::NavSatFix & message)
+  {
+    message.status.status = get_status();
+    message.status.service = get_service();
+    message.latitude = get_latitude();
+    message.longitude = get_longitude();
+    message.altitude = get_altitude();
+
+    if constexpr (use_covariance)
+    {
+      message.position_covariance = get_covariance();
+    }
+
+    return true;
+  }
+
+private:
+  struct Empty
+  {
+  };
+  using CovarianceArray = std::conditional_t<use_covariance, std::array<double, 9>, Empty>;
+  CovarianceArray covariance_;
```

Comment:
```suggestion
  std::array<double, 9> covariance_;
```

Why not simply use it this way?. I know that you are trying to make the variable covariance_ only exist when the `use_covariance` is set. Makes sense, but this `struct Empty` is a bit ugly. I think we can leave with it by default. Moreover, as it is in private scope it shouldn't be a problem at all. 

I don't have a strong opinion here, as you prefer

Reply:
Yep, it's not pretty for me either. I've updated it.

Reply:
Thank you 🫡